File: rs/cli/src/pe1.rs
extern crate pe;

use std::num::ParseIntError;

pub fn solve(args: Vec<String>) -> Result<(), ParseIntError> {
    let argc = args.len();

    let mut n = 1000;
    let mut is_loop = false;
    if argc > 1 {
        let arg1 = &args[1];
        if arg1 == "-l" {
            is_loop = true;
            if argc > 2 {
                let arg2 = &args[2];
                n = arg2.parse()?;
            }
        } else {
            n = arg1.parse()?;
        }
    }

    if is_loop {
        println!("n = {} (with loop)", n);
        println!("{}", pe::pe1::with_loop(n));
    } else {
        println!("n = {}", n);
        println!("{}", pe::pe1::no_loop(n));
    }

    Ok(())
}

File: cpp/cli/src/pe1.cc
#include "../include/pe1.h"

#include <cstdlib>
#include <cstring>
#include <iostream>

int pe1(int argc, char **argv)
{
  using namespace std;

  uint32_t n = 1000;
  bool is_loop = false;
  if (argc > 1) {
    if (strcmp(argv[1], "-l") == 0) {
      is_loop = true;
      if (argc > 2) {
        char *end;
        n = strtoul(argv[2], &end, 10);
      }
    } else {
      char *end;
      n = strtoul(argv[1], &end, 10);
    }
  }

  cout <<
    "n = " << n <<
    (is_loop ? " (with loop)" : "") << endl;
  cout <<
    (is_loop ? pe::with_loop(n) : pe::no_loop(n)) << endl;

  return 0;
}
